pkg/storage/azure/azure.go

```diff
@@ -75,47 +77,47 @@ func (e *errDoesNotExist) Error() string {
 	return e.Err.Error()
 }
 
-func getAzureConfigFromCloudSecret(creds *corev1.Secret) (*Azure, error) {
```

Comment:
What's the point of merging these functions?

Reply:
I did not see the point of having one function to populate one struct.

Reply:
To be sure that we don't use a wrong secret.

Reply:
We had a bug about that in the past.

Reply:
But tests should level the problem